### PR TITLE
feat: delete integration

### PR DESCRIPTION
**Pull request by JedrzejJanasiak**
## Issue

https://gravitee.atlassian.net/browse/APIM-4521

## Description

PR containing changes required for executing DELETE operation on specified integration.
While deleting there is additional validation performed to make sure that none associated federated API exist.

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-wqxtggidru.chromatic.com)

### CODE REVIEW

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/integration/use_case/DeleteIntegrationUseCase.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    ApiQueryService apiQueryService;
+
+    public DeleteIntegrationUseCase(IntegrationCrudService integrationCrudService, ApiQueryService apiQueryService) {
+        this.integrationCrudService = integrationCrudService;
+        this.apiQueryService = apiQueryService;
+    }
+
+    public void execute(Input input) {
+        apiQueryService
+            .search(ApiSearchCriteria.builder().integrationId(input.integrationId).build(), null, null)
```
Comment (jgiovaresco) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/integration/use_case/DeleteIntegrationUseCase.java]: I would define the last parameter (ApiFieldFilter) to exclude definition and picture as we don't need them here

File: gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-rest/src/test/java/io/gravitee/rest/api/management/v2/rest/resource/integration/IntegrationResourceTest.java [hunk 2]
```diff
@@ -312,4 +317,36 @@ public void should_return_400_when_missing_body() {
             assertThat(response).hasStatus(HttpStatusCode.BAD_REQUEST_400);
         }
     }
+
+    @Nested
+    class DeleteIntegration {
```
Comment (jgiovaresco) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-rest/src/test/java/io/gravitee/rest/api/management/v2/rest/resource/integration/IntegrationResourceTest.java:322]: There is a test missing to check the permission ;)

Comment (JedrzejJanasiak) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-rest/src/test/java/io/gravitee/rest/api/management/v2/rest/resource/integration/IntegrationResourceTest.java:322]: good catch, added :)

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/integration/use_case/DeleteIntegrationUseCase.java (hunk 1 above)
Comment (JedrzejJanasiak) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/integration/use_case/DeleteIntegrationUseCase.java]: done

**Review by jgiovaresco — APPROVED** [review]

**Pull request merged**